addons/rating/models/mail_thread.py

```diff
@@ -23,7 +23,45 @@ def unlink(self):
         return result
 
     def _get_message_create_valid_field_names(self):
-        return super()._get_message_create_valid_field_names() | {'rating_ids'}
+        return super()._get_message_create_valid_field_names() | {"rating_id", "rating_value"}
+
+    def _message_create(self, values_list):
+        """Override of _message_create rather than message_post to be just after create
+        but before any post-processing."""
+        rating_id_by_message_index = (values.pop("rating_id", None) for values in values_list)
+        rating_value_by_message_index = (values.pop("rating_value", None) for values in values_list)
```

Comment:
Couldn't those two generators+the use of zip just be reduced to `enumerate` and direct access to value list? It looks overcomplicated?

```py
for i, message in enumerate(messages):
    rating = values_list[i].pop("rating_id", None)
    rating_value = values_list[i].pop("rating_value", None)
```

Reply:
They need to be popped before calling super and are used afterwards.